### PR TITLE
Provide: offset query param to skip n results

### PR DESCRIPTION
**Pull request by omani**
* Provide `offset` query param to be able to offset the query
  * difference to `?page` param is that `page` can only be used in combination with `limit`
  * but if a client wants to make a query but wants to skip the first 10 there is no way to specify this with the `page` and `limit` combination since for example a:
   ```
   GET http://localhost:8080/users?page=2&limit=10 HTTP/1.1
   content-type: application/json
   ```
   would return a query result of 20 items while showing the last 10 of them since we're on the second page.

   while a:
   ```
   GET http://localhost:8080/users?offset=10 HTTP/1.1
   content-type: application/json
   ```
   would show the rest by skipping the first 10.

  * also available in combination with `limit`. so this is possible, too:
   ```
   GET http://localhost:8080/users?offset=10&limit=2 HTTP/1.1
   content-type: application/json
   ```

   would result in skipping the first 10 items, but showing only the first 2 items of the rest.

### CODE REVIEW

**Comment by omani** [comment]
@rs are you interested in this? then I would try to satisfy the CI checks.

**Comment by omani** [comment]
see https://github.com/rs/rest-layer/pull/61 for the counterpart in rest-layer.

**Comment by rs** [comment]
Yes I am interested. See my review comments.

File: mongo.go [hunk 1]
```diff
@@ -201,9 +201,26 @@ func (m *Handler) Find(ctx context.Context, lookup *resource.Lookup, page, perPa
...
+	for iter.Next(&mItem) {
+		// Check if context is still ok before to continue
+		if err = ctx.Err(); err != nil {
+			// TODO bench this as net/context is using mutex under the hood
+			iter.Close()
+			return nil, err
+		}
+		tmp.Items = append(tmp.Items, newItem(&mItem))
+	}
+
```
Comment (rs) [inline mongo.go]: Please remove that from the PR. Most people don't want this info, and it is not acceptable to double the number of requests for any given `Find`.

**Comment by omani** [comment]
done

**Comment by rs** [comment]
Regarding the discussion on https://github.com/rs/rest-layer/pull/62, we should revisit this PR to only use `offset`/`limit`.

Also, please stop opening a new PR each time you push a change, it makes things very hard to follow and it is not the way PR should be used. Just push your changes on the same branch and the PR will be updated.

**Comment by smyrman** [comment]
>  Just push your changes on the same branch and the PR will be updated.

@omani, if when you also want/need to rewrite history, here are some Git advice for how to do that without starting a new PR. If you already know these things, please feel free to ignore it, as I am just trying to help, not to offend you. I will assumed that you pull from this repo, and configured your GitHub fork as my-remote:

```bash
$ git fetch && git rebase origin/master      # Rebase on top of latest origin/master
$ git fetch && git rebase -i origin/master   # Allows for squashing, rewording, reordering  and/or removal of commits.
$ git push -f my-remote HEAD:refs/heads/feature_ofset # Force push from HEAD to your existing branch.
```

As you are probably aware, you should always be careful with `-f` / `--force`, e.g. never do it on the `master` branch on public projects. For PRs, it's usually OK, or even recommended to ensure clean history.

I also found a more detailed reference in the wiki of another GitHub project if you are interested: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request.
